hpx/runtime/threads/thread_init_data.hpp

```diff
@@ -63,9 +63,15 @@ namespace hpx { namespace threads
             parent_phase = rhs.parent_phase;
 #endif
 #ifdef HPX_HAVE_APEX
-        // HPX_HAVE_APEX forces the HPX_HAVE_THREAD_DESCRIPTION
-        // and HPX_HAVE_THREAD_PARENT_REFERENCE settings to be on
-            apex_data = apex_new_task(description, parent_locality_id, parent_id );
+            // HPX_HAVE_APEX forces the HPX_HAVE_THREAD_DESCRIPTION
+            // and HPX_HAVE_THREAD_PARENT_REFERENCE settings to be on
+            if (description.kind() == description.data_type_description) {
+                apex_data = util::apex_new_task(description.get_description(),
+                        parent_id);
+            } else {
+                apex_data = util::apex_new_task(description.get_address(),
+                        parent_id);
+            }
```

Comment:
Could we wrap this initialization into a helper function? It's used all over the place now...

Reply:
Yeah, this changed because of other reasons which don't make sense anymore. Will change.